src/hypercube.c

```diff
@@ -259,7 +261,7 @@ ts_hypercube_calculate_from_point(Hyperspace *hs, Point *p)
 			 * range. If a slice already exists, use that slice's ID instead
 			 * of a new one.
 			 */
-			ts_dimension_slice_scan_for_existing(cube->slices[i]);
+			ts_dimension_slice_scan_for_existing_and_lock_found(cube->slices[i], LockTupleKeyShare);
```

Comment:
Same here as above. Why do we need to lock and why is `KEY SHARE` appropriate here when `SHARE` is used above.